cmd/nerdctl/run_linux_test.go

```diff
@@ -104,3 +104,114 @@ func TestRunUlimit(t *testing.T) {
 	base.Cmd("run", "--rm", "--ulimit", ulimit2, testutil.AlpineImage, "sh", "-c", "ulimit -Sn").AssertOutExactly("622\n")
 	base.Cmd("run", "--rm", "--ulimit", ulimit2, testutil.AlpineImage, "sh", "-c", "ulimit -Hn").AssertOutExactly("722\n")
 }
+
+func TestRunWithInit(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	t.Parallel()
+	base := testutil.NewBase(t)
+	imageName := "test-init-with-custom-init-binary"
+	dockerfile := `
+FROM golang:1.18
+ENV GO111MODULE=off
+RUN echo  '\
+package main\n\
+\n\
+import (\n\
+    "fmt"\n\
+    "os"\n\
+    "os/signal"\n\
+    "syscall"\n\
+)\n\
+\n\
+func main() {\n\
+\n\
+    sigs := make(chan os.Signal, 1)\n\
+    done := make(chan bool, 1)\n\
+    signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)\n\
+    go func() {\n\
+        sig := <-sigs\n\
+        fmt.Printf("signal: %d\\n", sig)\n\
+        os.Stdout.Sync()\n\
+        done <- true\n\
+    }()\n\
+    for {\n\
+\n\
+    }\n\
+}\n'>> main.go
+
+RUN go build -o main .
+
+ENTRYPOINT ["./main"]
+`
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+	base.Cmd("build", "-t", imageName, buildCtx).AssertOK()
+	base.Cmd("build", buildCtx, "-t", imageName).AssertOK()
+	containerName := imageName + "container"
+	base.Cmd("run", "--rm", "--init", "--init-path", "/temp-init/nerdctl-init", imageName, "--name", containerName).AssertOK()
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+	defer base.Cmd("kill", containerName).AssertOK()
+	base.Cmd("kill", "--signal=\"SIGTERM\"", containerName).AssertOK()
+	logsCMD := base.Cmd("logs", containerName)
+	logsCMD.Base.T.Helper()
+	result := logsCMD.Run()
+	stdoutContent := result.Stdout() + result.Stderr()
+	assert.Assert(logsCMD.Base.T, result.ExitCode == 0, stdoutContent)
+	assert.Equal(logsCMD.Base.T, strings.Contains(stdoutContent, "signal: 15"), true)
```

Comment:
kill send a `SIGKILL` so this should be  `signal: 9`

Reply:
So I send SIGTERM here https://github.com/containerd/nerdctl/pull/948/files#diff-7312cd37020aca19ab86ba2f16dc89e6cf1a5a0580eb6ec3dc2aa06b3cd1ccb9R155

Reply:
sorry didn't pay attention 👍